demos/coreMQTT/mqtt_demo_plaintext.c

```diff
@@ -0,0 +1,943 @@
+ */
+static topicFilterContext_t xTopicFilterContext[ mqttexampleTOPIC_COUNT ] =
+{
+    { mqttexampleTOPIC, MQTTSubAckFailure }
+};
+
+
+/**
+ * @brief Static buffer used to hold MQTT messages being sent and received.
+ */
+static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = mqttexampleSHARED_BUFFER_SIZE
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief This example uses MQTT APIs to send and receive MQTT packets over a
+ * TCP connection established using Secure Sockets. This example is single
+ * threaded and uses statically allocated memory. This example uses QOS0 and
+ * therefore does not implement any retransmission mechanism for Publish
+ * messages. This example runs for democonfigMQTT_MAX_DEMO_COUNT, if the
+ * connection to the broker goes down, the code tries to reconnect to the broker
+ * with an exponential backoff mechanism.
+ */
+int RunCoreMqttPlaintextDemo( bool awsIotMqttMode,
+                              const char * pIdentifier,
+                              void * pNetworkServerInfo,
+                              void * pNetworkCredentialInfo,
+                              const void * pNetworkInterface )
+{
+    uint32_t ulPublishCount = 0U, ulTopicCount = 0U;
+    const uint32_t ulMaxPublishCount = 5UL;
+    NetworkContext_t xNetworkContext = { 0 };
+    MQTTContext_t xMQTTContext;
+    MQTTStatus_t xMQTTStatus;
+    BaseType_t xDemoStatus;
+    TransportSocketStatus_t xNetworkStatus;
+    uint32_t ulDemoRunCount = 0;
+    BaseType_t xIsConnectionEstablished = pdFALSE;
+
+    /* Upon return, EXIT_SUCCESS will indicate a successful demo execution.
+     * EXIT_FAILURE will indicate some failures occurred during execution. The
+     * user of this demo must check the logs for any failure codes. */
+    int lReturnStatus = EXIT_SUCCESS;
+
+    /* Remove compiler warnings about unused parameters. */
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    ulGlobalEntryTimeMs = prvGetTimeMs();
+
+    for( ; ulDemoRunCount < democonfigMQTT_MAX_DEMO_COUNT; ulDemoRunCount++ )
+    {
+        /****************************** Connect. ******************************/
+
+        /* Attempt to connect to the MQTT broker. If connection fails, retry after
+         * a timeout. The timeout value will be exponentially increased until the
+         * maximum number of attempts are reached or the maximum timeout value is
+         * reached. The function returns a failure status if the TCP connection
+         * cannot be established to the broker after the configured number of
+         * attempts. */
+        xNetworkStatus = prvConnectToServerWithBackoffRetries( &xNetworkContext );
+
+        if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+        {
+            lReturnStatus = EXIT_FAILURE;
+            break;
```

Comment:
Should we be avoiding multiple `break` statements and instead use `if( lReturnStatus == EXIT_SUCCESS )` checks to be consistent with the coding style of other demos?

Reply:
How do you want to get out of the loop and cleanup the connection without a break?

Reply:
Do we want to break out of the loop? As the `democonfigDEMO_MAX_COUNT` is configurable, it can be set to `1` to achieve the same behavior of only attempting once, and cleaning up on failure.

Reply:
Per offline conversation we have decided to not loop at all for simplicity of demonstrating the API.